### PR TITLE
Fix unexpected return tag on non-functions

### PR DESCRIPTION
**Pull request by HO-COOH**
# Fix/Feature/Other

## Fix
A dirty fix to #210 

- [ ] Description of the fix

Check the special return cases regardless of the setting for including return types. If it is `false` and all the case index is `-1`, then the `param` array shall be empty, so it mapped to nothing. Otherwise set `param` to `[""]`. So it could be mapped to `return`, without the types included.

- [ ] Added unit test

I added this test to `config.Test.ts`.

**NOTE**:
**All your `test()` calls seems to have wrong order, should be `test(actual, expected)`, but you have `test(expected, actual)`**.
I was so confused at first.

### CODE REVIEW

**Comment by cschlosser** [comment]
Thank you for your contribution.
Could you please remove the package-lock.json.
Some tools get confused when there is a yarn lock file as well.

Regarding the order of arguments for assertions: This is probably due to the diff being backwards otherwise iirc.
I can take another look at it to see if this is something that has sneaked in.

File: src/Lang/Cpp/CppDocGen.ts [hunk 1]
```diff
@@ -264,9 +264,9 @@ export class CppDocGen implements IDocGen {
     }
 
     protected generateReturnParams(): string[] {
-        if (this.cfg.Generic.includeTypeAtReturn === false) {
-            return [""];
-        }
+        // if (this.cfg.Generic.includeTypeAtReturn === false) {
+        //     return [""];
+        // }
```
Comment (cschlosser) [inline src/Lang/Cpp/CppDocGen.ts:269]: Please don't just disable lines by turning them into comments.
Either remove them or give a reason why these lines are comments now.

**Comment by HO-COOH** [comment]
@cschlosser see this update

For the wrong test I talked about, this is what I see
image

**Comment by cschlosser** [comment]
You're totally right.
I don't know why but this seems to have changed.

**Pull request merged**